### PR TITLE
Remove deprecated alias macro(1/3)

### PR DESCRIPTION
**Pull request by fffrog**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #137562
* #137559
* __->__ #137556

**Detailed Descriptions:**
- Remove AT_ERROR Macro

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @mcarilli @ptrblck @leslie-fang-intel @EikanWang

### CODE REVIEW

File: c10/util/Exception.h [hunk 1]
```diff
@@ -740,14 +733,4 @@ inline void deprecated_AT_ASSERTM() {}
     C10_EXPAND_MSVC_WORKAROUND(TORCH_INTERNAL_ASSERT(cond, __VA_ARGS__)); \
   } while (false)
 
-// Deprecated alias; this alias was deprecated because it represents extra API
-// surface that makes it hard for people to understand what macro to use.
-// Use TORCH_CHECK(false, ...) or TORCH_INTERNAL_ASSERT(false, ...) to
-// unconditionally fail at a line of code.
-#define AT_ERROR(...)                                                        \
```
Comment (albanD) [inline c10/util/Exception.h:747]: I'm afraid there are too many users of this outside of PyTorch core to remove it right now: https://github.com/search?q=%2F%28%3F-i%29AT_ERROR%5C%28%2F&type=code

Please add this API back for now, I don't think the churn for users is worth the maintenance burden of this function at the moment.

**Review by albanD — COMMENTED** [review]
Thanks for the update!
Let's remove all use on our end but I'm afraid there are too many usage out of core to remove it right now. Or at least we should do it in aseparate PR that can easily be reverted.

File: c10/util/Exception.h (hunk 1 above)
Comment (FFFrog) [reply c10/util/Exception.h:747]: I agree with you totally.
There are so many usages of the three API out of PyTorch core, so it`s not the right time to remove those, I will close those three PR now, thank you for your review.

Comment (FFFrog) [reply c10/util/Exception.h:747]: Sure, I will update the pr right now :D

**Comment by FFFrog** [comment]
Hi @ezyang, please have to take a look at this pr, thank you.

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**